ocs_ci/utility/managedservice.py

```diff
@@ -92,29 +92,48 @@ def remove_header_footer_from_key(key):
     return "".join(key_lines)
 
 
-def get_storage_provider_endpoint(cluster):
+def get_storage_provider_endpoint(wait=False, timeout=1080):
     """
-    Get get_storage_provider_endpoint
+    Get storage provider endpoint from Provider storage cluster or from
+    configuration if configured.
 
     Args:
-        cluster (str): cluster name
+        wait (bool): If true then wait for the value to be available for
+            number of seconds defined in timeout parameter. If false then
+            trye to return value only once.
+        timeout (int): Number of seconds to wait for the value
 
     Returns:
         str: value of storage provider endpoint
 
     """
+    provider_endpoint = config.DEPLOYMENT.get("storage_provider_endpoint")
+    if provider_endpoint:
+        logger.info(
+            f"Provider endpoint was loaded from configuration: {provider_endpoint}"
+        )
+        return provider_endpoint
+
+    config.switch_to_provider()
+
+    def _get_provider_endpoint():
+        oc = ocp.OCP(namespace="openshift-config")
+        oc.exec_oc_cmd(
+            "get storagecluster -n openshift-storage -o=jsonpath="
+            "'{.items[0].status.storageProviderEndpoint}'"
+        )
 
-    # TODO: p2 task to implement below functionality
-    #  Use multicluster implementation to use
-    #  kubeconfig as per cluster name and
-    #  extract value of storage_provider_endpoint
-    #  handle invalid cluster name in implementation
-    #  validate Return String storage provider endpoint:
-    #     1. raise Error if storage_provider_endpoint is
-    #        not found in cluster yaml
-    #     2. warning if storage cluster is not ready
-    #     and storage_provider_endpoint is available in
-    #     storagecluster yaml .
-    #  For now use hardcoded value from config with key
-    #  storage_provider_endpoint:
-    return config.DEPLOYMENT.get("storage_provider_endpoint", "")
+    try:
+        if wait:
+            for result in TimeoutSampler(
+                timeout=timeout, sleep=5, func=_get_provider_endpoint
+            ):
+                if result:
+                    provider_endpoint = result
+                    break
+        else:
+            provider_endpoint = _get_provider_endpoint()
+    finally:
+        config.reset_ctx()
```

Comment:
Why do we need this method here? If we want the original cluster, I think we had better save the original cluster index and use the method `switch_ctx` when passing that index.

Reply:
If `reset_ctx` works then I would prefer it to longer code construct that saves current context id and switches it back. I added https://github.com/red-hat-storage/ocs-ci/issues/5831 to better document those methods.